R/as_tibble.R

```diff
@@ -116,6 +117,8 @@ as_tibble.data.frame <- function(x, validate = NULL, ...,
 as_tibble.list <- function(x, validate = NULL, ..., .rows = NULL,
                            .name_repair = c("check_unique", "unique", "universal", "minimal")) {
 
+  signal_soft_deprecated("`as_tibble.list()` is deprecated, use `tibble()` or `new_tibble()` instead.")
```

Comment:
I'm worried that it's going to break a lot of existing code; I think it would be better to "retire" this function. i.e. we'll keep it around for a long time, but need to update the documentation to suggest alternatives.